WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -177,37 +215,49 @@ private MediaModel getNextMediaToUpload() {
     }
 
     private void addUniqueMediaToQueue(MediaModel media) {
-        for (MediaModel queuedMedia : getUploadQueue()) {
-            if (queuedMedia.getLocalSiteId() == media.getLocalSiteId() &&
-                    StringUtils.equals(queuedMedia.getFilePath(), media.getFilePath())) {
-                return;
+        if (media != null) {
+            for (MediaModel queuedMedia : getUploadQueue()) {
+                if (queuedMedia.getLocalSiteId() == media.getLocalSiteId() &&
+                        StringUtils.equals(queuedMedia.getFilePath(), media.getFilePath())) {
+                    return;
+                }
             }
-        }
 
-        // no match found in queue
-        getUploadQueue().add(media);
+            // no match found in queue
+            getUploadQueue().add(media);
+        }
     }
 
     private void unpackIntent(@NonNull Intent intent) {
-        mSite = (SiteModel) intent.getSerializableExtra(WordPress.SITE);
-
-        // add local queued media from store
-        List<MediaModel> localMedia = mMediaStore.getLocalSiteMedia(mSite);
-        if (localMedia != null && !localMedia.isEmpty()) {
-            // uploading is updated to queued, queued media added to the queue, failed media added to completed list
-            for (MediaModel mediaItem : localMedia) {
-                if (MediaUploadState.UPLOADING.name().equals(mediaItem.getUploadState())) {
-                    mediaItem.setUploadState(MediaUploadState.QUEUED.name());
-                    mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(mediaItem));
-                }
 
-                if (MediaUploadState.QUEUED.name().equals(mediaItem.getUploadState())) {
-                    addUniqueMediaToQueue(mediaItem);
-                } else if (MediaUploadState.FAILED.name().equals(mediaItem.getUploadState())) {
-                    getCompletedItems().add(mediaItem);
-                }
-            }
-        }
+        SiteModel site = (SiteModel) intent.getSerializableExtra(WordPress.SITE);
+        long postId = intent.getLongExtra(POST_ID_KEY, 0);
+
+        // TODO right now, in the case we had pending uploads and the app/service was restarted,
```

Comment:
also removed SiteModel key as a bundle in the Intent as it was not being used - 2359e87